12/umbraco-cms/reference/content-delivery-api/extension-points/extension-api-for-querying.md

```diff
@@ -0,0 +1,308 @@
+```
+
+The response will include the blog posts associated with the provided authors' IDs, enabling us to get only the relevant results from the API.
+
+**Response**
+
+```json
+{
+    "total": 4,
+    "items": [
+        ...
+    ]
+}
+```
+
+Our filter implementation follows a similar structure to the custom selector we discussed earlier. We continue to utilize the `IContentIndexHandler` interface, but this time we introduce the `IFilterHandler`. This combination gives us flexibility and control over the filtering behaviour.
+
+The underlying idea remains consistent: we store the author's `Guid` in a new `"authorId"` field within the index. Consequently, we will need to rebuild the index to reflect the changes.
+
+To illustrate the implementation, consider the following code example:
+
+{% code title="AuthorFilter.cs" lineNumbers="true" %}
+```csharp
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.DeliveryApi;
+using Umbraco.Cms.Core.Models;
+
+namespace Umbraco.Docs.Samples;
+
+public class AuthorFilter : IFilterHandler, IContentIndexHandler
+{
+    private const string AuthorSpecifier = "author:";
+    private const string FieldName = "authorId";
+
+    // Querying
+    public bool CanHandle(string query)
+        => query.StartsWith(AuthorSpecifier, StringComparison.OrdinalIgnoreCase);
+
+    public FilterOption BuildFilterOption(string filter)
+    {
+        var fieldValue = filter.Substring(AuthorSpecifier.Length);
+
+        // There might be several values for the filter
+        var values = fieldValue.Split(',');
+
+        return new FilterOption
+        {
+            FieldName = FieldName,
+            Values = values,
+            Operator = FilterOperation.Is
+        };
+    }
+
+    // Indexing
+    public IEnumerable<IndexFieldValue> GetFieldValues(IContent content, string? culture)
+    {
+        GuidUdi? authorUdi = content.GetValue<GuidUdi>("author");
+
+        if (authorUdi is null)
+        {
+            return Array.Empty<IndexFieldValue>();
+        }
+
+        return new[]
+        {
+            new IndexFieldValue
+            {
+                FieldName = FieldName,
+                Values = new object[] { authorUdi.Guid }
+            }
+        };
+    }
+
+    public IEnumerable<IndexField> GetFields() => new[]
+    {
+        new IndexField
+        {
+            FieldName = FieldName,
+            FieldType = FieldType.StringRaw,
+            VariesByCulture = false
+        }
+    };
+}
+```
+{% endcode %}
+
+This filter performs an exact match for each specified GUID against the index. The Delivery API also supports "Contains" filters, which can be used, for instance, to create an author filter based on names instead of IDs. In this case, the `BuildFilterOption()` method should use the `FilterOperation.Contains` operator, and the `GetFields()` implementation should specify `FieldType.StringAnalyzed` for the returned `IndexField`.
```

Comment:
```suggestion
The principal difference from the selector is that the filter implements `BuildFilterOption()` instead of `BuildSelectorOption()`. Here, the filter performs an exact match for any specified `Guid` in the query. Efficiently this makes the filter perform an `OR` operation against the index. 

Since we need to perform an exact match, the index field (`authorId`) is once again defined as a "raw" string. Other options include "analyzed" and "sortable" strings. These support "contains" searches and alpha-numeric sorting, respectively.
```